packages/dds/sequence/src/test/intervalRevertibles.fuzz.spec.ts

```diff
@@ -0,0 +1,376 @@
+	 * Note due to concurency, during test execution the actual number of intervals may exceed this.
+	 */
+	maxIntervals?: number;
+	maxInsertLength?: number;
+	intervalCollectionNamePool?: string[];
+	propertyNamePool?: string[];
+	validateInterval?: number;
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+	maxStringLength: 1000,
+	maxIntervals: 100,
+	maxInsertLength: 10,
+	intervalCollectionNamePool: ["comments"],
+	propertyNamePool: ["prop1", "prop2", "prop3"],
+	validateInterval: 100,
+};
+
+const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
+const revertibles: SharedStringRevertible[] = [];
+
+emitter.on("clientCreate", (client) => {
+	const string = client.channel as SharedString;
+	const labels = string.getIntervalCollectionLabels();
+	Array.from(labels).forEach((label) => {
+		const collection = string.getIntervalCollection(label);
+		collection.on("addInterval", (interval, local, op) => {
+			appendAddIntervalToRevertibles(interval, revertibles);
+		});
+		// Note: delete and change interval edits are disabled for now, and will be reenabled
+		// once bugs AB#4544 and AB#4543 (respectively) are resolved.
+
+		// collection.on("deleteInterval", (interval, local, op) => {
+		// 	if (local) {
+		// 		appendDeleteIntervalToRevertibles(string, interval, revertibles);
+		// 	}
+		// });
+		// collection.on("changeInterval", (interval, previousInterval, local, op) => {
+		// 	if (local) {
+		// 		appendChangeIntervalToRevertibles(string, interval, previousInterval, revertibles);
+		// 	}
+		// });
+		collection.on("propertyChanged", (interval, propertyDeltas, local, op) => {
+			if (local) {
+				appendIntervalPropertyChangedToRevertibles(interval, propertyDeltas, revertibles);
+			}
+		});
+		string.on("sequenceDelta", (op) => {
+			if (op.isLocal) {
+				appendSharedStringDeltaToRevertibles(string, op, revertibles);
+			}
+		});
+	});
+});
+
+const intervalTestOptions = {
+	validationStrategy: { type: "fixedInterval", interval: 10 },
+	reconnectProbability: 0.1,
+	numberOfClients: 3,
+	clientJoinOptions: {
+		maxNumberOfClients: 6,
+		clientAddProbability: 0.1,
+	},
+	defaultTestCount: 100,
+	// Uncomment this line to replay a specific seed from its failure file:
+	// replay: 0,
+	saveFailures: { directory: path.join(__dirname, "../../src/test/results") },
+	parseOperations: (serialized: string) => {
+		const operations: Operation[] = JSON.parse(serialized);
+		// Replace this value with some other interval ID and uncomment to filter replay of the test
+		// suite to only include interval operations with this ID.
+		// const filterIntervalId = "00000000-0000-0000-0000-000000000000";
+		// if (filterIntervalId) {
+		// 	return operations.filter((entry) =>
+		// 		[undefined, filterIntervalId].includes((entry as any).id),
+		// 	);
+		// }
+		return operations;
+	},
+};
+
+const optionsWithEmitter = {
+	...intervalTestOptions,
+	emitter,
+};
+
+type ClientOpState = FuzzTestState;
+function makeOperationGenerator(
```

Comment:
you should be able to avoid duplicating all of this code by making your own generator which:

- delegates to the base generator (everything except revertibles-specific code) with some weight
- generates a revertible with some other weight

Reply:
since you want to disable add/delete, to do this you'll need to make the weights on those operations configurable. this is a good change I've been meaning to make anyway.